msprime/stats.py

```diff
@@ -581,13 +583,28 @@ def tree_stat_windowed(self, sample_sets, weight_fun, windows=None):
         assert len(out[0]) == 1
         return [x[0] for x in out]
 
+
+class BranchLengthStatCalculator(GeneralStatCalculator):
+    """
+    Class for calculating a broad class of tree statistics.  These are all
+    calculated using :meth:``BranchLengthStatCalculator.tree_stat_vector`` as the
+    underlying engine.  This class requires the `numpy
+    <http://www.numpy.org/>`_ library.
+
+    NOTE: this interface is still in beta, and may change in the future.
```

Comment:
Let's put in a sphinx [..warning](http://www.sphinx-doc.org/en/stable/markup/para.html) in here (and the other classes) to make sure it's very obviously a warning.